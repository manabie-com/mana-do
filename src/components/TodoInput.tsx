import React, { useRef } from 'react'

interface Props{
    onCreateTodo: any;
    inputRef: any
}

const TodoInput: React.FC<Props> = ({onCreateTodo, inputRef}) => {
    return (
        <div className="Todo__creation">
        <input
            ref={inputRef}
            className="Todo__input"
            placeholder="What need to be done?"
            onKeyDown={onCreateTodo}
        />
    </div>
    )
}

export default TodoInput;