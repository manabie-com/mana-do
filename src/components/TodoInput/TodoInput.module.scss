.inputContainer {
	display: flex;
	padding: 0 24px;
}

.checkDecor {
	width: 20px;
	height: 20px;
	background-color: #e2e2e2;
	border: 3px solid #e2e2e2;
	border-radius: 9px;
	margin-right: 10px;
}

.input {
	border: none;
	background-color: transparent;
	width: 100%;
	outline: none;
	line-height: 1.8;
	font-size: 14px;

	&::placeholder {
		color: rgba(65, 63, 63, 0.6);
	}
}
