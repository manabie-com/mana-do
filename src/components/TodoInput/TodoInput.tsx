import React, { useState } from "react";

import styles from "./TodoInput.module.scss";

const defaultValue = "";

export default function TodoInput({ addTodo }: any): JSX.Element {
  const [value, setValue] = useState(defaultValue);

  const handleKeyDown = (event: any) => {
    if (event.key === "Enter" && Boolean(value)) {
      console.log("dispatch: ", value);
      setValue(defaultValue);
      addTodo(value);
    }
  };

  return (
    <div className={styles.inputContainer}>
      <input
        aria-label="input"
        className={styles.input}
        placeholder="What need to be done?"
        autoFocus
        value={value}
        onChange={(e) => setValue(e.target.value)}
        onKeyDown={handleKeyDown}
      />
    </div>
  );
}
