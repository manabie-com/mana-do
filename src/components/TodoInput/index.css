.Todo__creation {
  display: flex;
}

.Todo__input {
  flex: 1 1;
}

.input-todo {
  display: flex;
}
