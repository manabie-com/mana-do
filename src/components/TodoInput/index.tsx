import React, { useRef } from "react";

export interface TodoInputProps {
  defaultValue?: string;
  onEnter?: (value: string) => void;
  autoFocus?: boolean;
}

const TodoInput: React.FC<TodoInputProps> = ({
  defaultValue = "",
  autoFocus = false,
  onEnter,
}) => {
  const inputRef = useRef<HTMLInputElement>(null);

  const handleKeyDown = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter" && inputRef.current) {
      onEnter && onEnter(inputRef.current.value);
      inputRef.current.value = "";
    }
  };

  return (
    <input
      ref={inputRef}
      defaultValue={defaultValue}
      className="Todo__input"
      placeholder="What need to be done?"
      onKeyDown={handleKeyDown}
      autoFocus={autoFocus}
    />
  );
};

export default TodoInput;
