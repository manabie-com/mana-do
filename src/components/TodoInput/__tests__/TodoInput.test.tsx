import React from "react";
import { expect, test, describe } from "@jest/globals";
import { fireEvent, render, screen } from "@testing-library/react";
import TodoInput, { TodoInputProps } from "../index";

const mockedOnEnter = jest.fn();

describe("TodoInput snapshots", () => {
  test("should render with empty text", () => {
    const props: TodoInputProps = {
      defaultValue: "",
      onEnter: mockedOnEnter,
      autoFocus: false,
    };
    const { container } = render(<TodoInput {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render with default text", () => {
    const props: TodoInputProps = {
      defaultValue: "Do homework",
      onEnter: mockedOnEnter,
      autoFocus: false,
    };
    const { container } = render(<TodoInput {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("TodoInput functions", () => {
  test("should have correct default value of input", () => {
    const props: TodoInputProps = {
      defaultValue: "Do homework",
      onEnter: mockedOnEnter,
      autoFocus: false,
    };
    render(<TodoInput {...props} />);
    const inputElement = screen.getByRole("textbox") as HTMLInputElement;
    expect(inputElement.value).toBe("Do homework");
  });

  test("should be able to type in input", () => {
    const props: TodoInputProps = {
      defaultValue: "",
      onEnter: mockedOnEnter,
      autoFocus: false,
    };
    render(<TodoInput {...props} />);
    const inputElement = screen.getByRole("textbox") as HTMLInputElement;
    fireEvent.change(inputElement, { target: { value: "Go to market" } });
    expect(inputElement.value).toBe("Go to market");
  });

  test("should have empty value when press Enter", () => {
    const props: TodoInputProps = {
      defaultValue: "",
      onEnter: mockedOnEnter,
      autoFocus: false,
    };
    render(<TodoInput {...props} />);
    const inputElement = screen.getByRole("textbox") as HTMLInputElement;
    fireEvent.change(inputElement, { target: { value: "Go to market" } });
    fireEvent.keyDown(inputElement, { key: "Enter" });
    expect(inputElement.value).toBe("");
  });

  test("should focus when autofocus is true", () => {
    const props: TodoInputProps = {
      defaultValue: "",
      onEnter: mockedOnEnter,
      autoFocus: true,
    };
    render(<TodoInput {...props} />);
    const inputElement = screen.getByRole("textbox") as HTMLInputElement;
    expect(document.activeElement).toEqual(inputElement);
  });
});
