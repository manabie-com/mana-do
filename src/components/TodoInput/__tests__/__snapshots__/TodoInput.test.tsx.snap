// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput snapshots should render with default text 1`] = `
<input
  class="Todo__input"
  placeholder="What need to be done?"
  value="Do homework"
/>
`;

exports[`TodoInput snapshots should render with empty text 1`] = `
<input
  class="Todo__input"
  placeholder="What need to be done?"
  value=""
/>
`;
